subprojects/build-cache-packaging/src/main/java/org/gradle/caching/internal/packaging/impl/PermissionUtils.java

```diff
@@ -0,0 +1,139 @@
+package org.gradle.caching.internal.packaging.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.attribute.DosFileAttributeView;
+import java.nio.file.attribute.DosFileAttributes;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.EnumMap;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Helps dealing with file permissions.
+ */
+public class PermissionUtils {
+
+    /**
+     * XXX: When using standard Java permissions, we treat 'owner' and 'group' equally and give no
+     * permissions for 'others'.
+     */
+    private enum StandardFilePermission {
+        EXECUTE(0110), WRITE(0220), READ(0440);
+
+        private int mode;
+
+        StandardFilePermission(int mode) {
+            this.mode = mode;
+        }
+    }
+
+    private static Map<PosixFilePermission, Integer> posixPermissionToInteger = new EnumMap<>(PosixFilePermission.class);
+
+    static {
+        posixPermissionToInteger.put(PosixFilePermission.OWNER_EXECUTE, 0100);
+        posixPermissionToInteger.put(PosixFilePermission.OWNER_WRITE, 0200);
+        posixPermissionToInteger.put(PosixFilePermission.OWNER_READ, 0400);
+
+        posixPermissionToInteger.put(PosixFilePermission.GROUP_EXECUTE, 0010);
+        posixPermissionToInteger.put(PosixFilePermission.GROUP_WRITE, 0020);
+        posixPermissionToInteger.put(PosixFilePermission.GROUP_READ, 0040);
+
+        posixPermissionToInteger.put(PosixFilePermission.OTHERS_EXECUTE, 0001);
+        posixPermissionToInteger.put(PosixFilePermission.OTHERS_WRITE, 0002);
+        posixPermissionToInteger.put(PosixFilePermission.OTHERS_READ, 0004);
+    }
+
+    public static void setPermissions(File file, int mode) throws IOException {
+        if (IS_POSIX) {
+            Files.setPosixFilePermissions(file.toPath(), PermissionUtils.permissions(mode));
```

Comment:
Is it necessary to qualify this call with `PermissionUtils.`?

Reply:
Removed it